sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,185 @@
+      const [file, ...parameters] = command;
+      const buffer = await processUtils.execFile(file, parameters);
+      results.push(buffer.toString("utf-8"));
+    }
+  } catch (e) {
+    throw new Error(e.toString("utf-8"));
+  }
+
+  return results;
+}
+
+/**
+ * Known PowerShell errors
+ * @internal
+ */
+export const powerShellErrors = {
+  login: "Run Connect-AzAccount to login",
+  installed:
+    "The specified module 'Az.Accounts' with version '2.2.0' was not loaded because no valid module file was found in any module directory"
+};
+
+/**
+ * Messages to use when throwing in this credential.
+ * @internal
+ */
+export const powerShellPublicErrorMessages = {
+  login:
+    "Please run 'Connect-AzAccount' from powershell to authenticate before using this credential.",
+  installed: `The 'Az.Account' module >= 2.2.0 is not installed. Install the Azure Az PowerShell module with: "Install - Module - Name Az - Scope CurrentUser - Repository PSGallery - Force".`
+};
+
+// PowerShell Azure User not logged in error check.
+const isLoginError = (err: Error) => err.message.match(`(.*)${powerShellErrors.login}(.*)`);
+
+// Az Module not Installed in Azure PowerShell check.
+const isNotInstallError = (err: Error) => err.message.match(powerShellErrors.installed);
+
+/**
+ * Optional parameters for the {@link AzurePowerShellCredential} class.
+ */
+export interface AzurePowerShellCredentialOptions {
+  /**
+   * If specified, this credential will use the legacy `powershell` command instead of the newest `pwsh`.
+   */
+  useLegacyPowerShell?: boolean;
+}
+
+/**
+ * This credential will use the currently logged-in user login information via the Azure Power Shell command line tool.
+ * To do so, it will read the user access token and expire time with Azure Power Shell command `Get-AzAccessToken -ResourceUrl {ResourceScope}`.
+ * To be able to use this credential, ensure that:
+ * - Install the Azure Az PowerShell module with: `Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force`.
+ * - You have already logged in via the 'az' tool using the command "Connect-AzAccount" from the command line.
+ */
+export class AzurePowerShellCredential implements TokenCredential {
+  private readonly powerShellCommand: string;
+
+  /**
+   * Creates an instance of the ClientCertificateCredential with the details
+   * needed to authenticate against Azure Active Directory with a certificate.
+   *
+   * @param useLegacyPowerShell - The flag indicating if legacy powershell should be used for authentication.
+   */
+  constructor(options?: AzurePowerShellCredentialOptions) {
+    if (options?.useLegacyPowerShell) {
+      this.powerShellCommand = formatCommand("powershell");
+    } else {
+      this.powerShellCommand = formatCommand("pwsh");
+    }
+  }
+  /**
+   * Gets the access token from Azure Power Shell
+   * @param resource - The resource to use when getting the token
+   */
+  private async getAzurePowerShellAccessToken(
```

Comment:
we should test the credential with Legacy powershell, to ensure the commands are working fine with it.

Reply:
Thank you! Turns out `-v` won't work. I'm testing the other commands on legacy

Reply:
`Get-AzAccessToken -ResourceUrl` didn't work on legacy. I sent you the error over Teams. It could be an issue with my local environment.

Reply:
It doesn't fail! I typed the command wrong.

Reply:
Alright, everything is working with the legacy powershell 👍